File: sol/2302.cpp
struct Solution {
    using ll = long long;
    long long countSubarrays(vector<int>& nums, long long k) {
        ll res = 0, partSum = 0;
        int s = 0, n = nums.size();
        for (int e = 0; e < n; ++e){
            partSum += nums[e];
            while (partSum * (e - s + 1) >= k){
                partSum -= nums[s++];
            }
            res += e - s + 1;
        }
        return res;
    }
};

File: sol/2302/2302.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        res = 0
        s = 0
        n = len(nums)
        partSum = 0
        for e in range(n):
            partSum += nums[e]
            while partSum * (e - s + 1) >= k:
                partSum -= nums[s]
                s += 1
            res += e - s + 1
        return res
